### PR TITLE
Add migration on 3box imports and remove feature flag

### PR DESCRIPTION
**Pull request by danjm**
This PR makes two notable changes:
- Updates the threebox controller so that data retrieved from threebox is run through new migrations before it is restored to metamask
- removes the 3box feature flag

### CODE REVIEW

File: app/scripts/controllers/threebox.js [hunk 1]
```diff
@@ -73,12 +71,19 @@ class ThreeBoxController {
...
       if (threeBoxSyncingAllowed && threeBoxSynced) {
-        await this.space.private.set('lastUpdated', Date.now())
-        await this.space.private.set(type, JSON.stringify(newState))
+        const newState = {
+          preferences: this.preferencesController.store.getState(),
+          addressBook: this.addressBookController.state,
+          lastUpdated: Date.now(),
+          lastMigration: this.lastMigration,
+        }
+        newState[type] = newTypeState
```
Comment (Gudahtt) [inline app/scripts/controllers/threebox.js]: I don't think we need this line anymore, as we're setting the most current address book and preferences state above anyway.

File: app/scripts/controllers/threebox.js [hunk 2]
```diff
@@ -103,13 +110,23 @@ class ThreeBoxController {
...
+    this.address = await this.keyringController.getAppKeyAddress(accounts[0], 'wallet://3box.metamask.io')
+    let backupExists
+    try {
+      const threeBoxConfig = await Box.getConfig(this.address)
+      backupExists = threeBoxConfig.spaces && threeBoxConfig.spaces.metamask
+    } catch (e) {
+      log.error(e)
+      backupExists = false
+    }
+    if (this.getThreeBoxSyncingState() || backupExists) {
```
Comment (Gudahtt) [inline app/scripts/controllers/threebox.js]: Is `this.getThreeBoxSyncingState()` redundant in this case? I'm not sure it's possible to get to this point unless it's truthy

File: app/scripts/controllers/threebox.js [hunk 3]
```diff
@@ -103,13 +110,23 @@ class ThreeBoxController {
     })
   }
 
+  async _checkThreeBoxBackupExistance () {
```
Comment (Gudahtt) [inline app/scripts/controllers/threebox.js]: I'm guessing this was an accident

File: app/scripts/controllers/threebox.js (hunk 2 above)
Comment (danjm) [reply app/scripts/controllers/threebox.js]: Not redundant, as `initializeThreeBox` is called on import

**Review by Gudahtt — DISMISSED** [review]
Looks good!

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**